contrib/dracut/90zfs/zfs-generator.sh.in

```diff
@@ -15,8 +15,10 @@ type getarg >/dev/null 2>&1 || . "$dracutlib"
 [ "${root##zfs:}" = "${root}" -a "${root##ZFS=}" = "${root}" -a "$rootfstype" != "zfs" ] && exit 0
 
 rootfstype=zfs
-if echo "${rootflags}" | grep -q zfsutil ; then
+if echo "${rootflags}" | grep -Eq '^zfsutil$|^zfsutil,|,zfsutil$|,zfsutil,' ; then
     true
+elif test -n "${rootflags}" ; then
+    rootflags="zfsutil,${rootflags}"
 else
     rootflags=zfsutil
 fi
```

Comment:
I would like that to be the case, but, unfortunately for us, this is far too early in the boot process to be running anything like find_bootfs.  Unlike during the execution of mount-zfs.sh, the pools are not imported at this time, you gotta understand, this is the EARLIEST part of the boot process.  And, after the generators have executed, there is no way to alter the mount unit any further.

So, it looks like there is no way to support that feature (at least not without major rethinking of how it works).  The only way I see to make zfs:AUTO work, would be to require the execution of mount-zfs.sh unconditionally, as well as to require the complete suppression of sysroot.mount — and who knows what other hellish effects that may have!

To be honest, it's not a big deal for people who are using https://github.com/Rudd-O/zfs-fedora-installer/tree/master/grub-zfs-fixer — the root dataset and mountpoint are properly deduced from /etc/fstab in that case, so there is zero need to specify anything on any config file, as the right thing happens upon generation of the grub.cfg file.

I await until brighter minds find a solution to this problem.